inference-engine/src/transformations/src/transformations/convert_nms_4_to_nms_dynamic.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/graph_util.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "transformations/convert_nms_4_to_nms_dynamic.hpp"
+
+bool ngraph::pass::UpgradeNMS4ToNMSDynamic::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    bool rewritten = false;
+    if (!ngraph::op::util::has_op_with_type<ngraph::opset4::NonZero>(f)) {
+        return rewritten;
+    }
+
+    for (auto &node : f->get_ops()) {
+        auto nms_4 = ngraph::as_type_ptr<opset4::NonMaxSuppression>(node);
```

Comment:
Please use `GraphRewrite` here. You can override `run_on_functoin` method to check that function has NonZero op.

Reply:
Done